app/backend/src/couchers/models.py

```diff
@@ -653,3 +653,268 @@ class InitiatedUpload(Base):
+    cluster_id = Column(ForeignKey("clusters.id"), nullable=False, index=True)
+    role = Column(Enum(GroupRole), nullable=False)
+    joined = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+    left = Column(DateTime(timezone=True))
+
+
+class ClusterPageAssociation(Base):
+    """
+    pages related to clusters
+    """
+
+    __tablename__ = "cluster_page_associations"
+    __table_args__ = (UniqueConstraint("page_id", "cluster_id"),)
+
+    id = Column(BigInteger, primary_key=True)
+
+    page_id = Column(ForeignKey("pages.id"), nullable=False, index=True)
+    cluster_id = Column(ForeignKey("clusters.id"), nullable=False, index=True)
+
+
+class PageType(enum.Enum):
+    main_page = 1
+    point_of_interest = 2
+    guide = 3
+    event = 4
+
+
+class Page(Base):
+    """
+    similar to a wiki page about a community, POI or guide
+    """
+
+    __tablename__ = "pages"
+
+    id = Column(BigInteger, primary_key=True)
+
+    type = Column(Enum(PageType), nullable=False)
+    thread_id = Column(ForeignKey("threads.id"), nullable=False, unique=True, index=True)
+    creator_user_id = Column(ForeignKey("users.id"), nullable=False, index=True)
+    owner_user_id = Column(ForeignKey("users.id"), index=True)
+    owner_cluster_id = Column(ForeignKey("clusters.id"), unique=True, index=True)
+
+    # Only one of owner_user and owner_cluster should be set
+    CheckConstraint(
+        "owner_user_id IS NULL AND owner_cluster_id IS NOT NULL OR owner_user_id IS NOT NULL AND owner_cluster_id IS NULL",
+        name="one_owner",
+    )
+
+
+class PageVersion(Base):
+    """
+    version of page content
+    """
+
+    __tablename__ = "pageversions"
+
+    id = Column(BigInteger, primary_key=True)
+
+    page_id = Column(ForeignKey("pages.id"), nullable=False, index=True)
+    editor_user_id = Column(ForeignKey("users.id"), nullable=False, index=True)
+    title = Column(String, nullable=False)
+    content = Column(String, nullable=False)
+    geom = Column(Geometry(geometry_type="POINT", srid=4326), nullable=False)
+    created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+
+
+class ClusterEventAssociation(Base):
+    """
+    events related to clusters
+    """
+
+    __tablename__ = "cluster_event_associations"
+    __table_args__ = (UniqueConstraint("event_id", "cluster_id"),)
+
+    id = Column(BigInteger, primary_key=True)
+
+    event_id = Column(ForeignKey("events.id"), nullable=False, index=True)
+    cluster_id = Column(ForeignKey("clusters.id"), nullable=False, index=True)
+
+
+class Event(Base):
+    """
+    A happening
+    """
+
+    __tablename__ = "events"
+
+    id = Column(BigInteger, primary_key=True)
+
+    title = Column(String, nullable=False)
+    content = Column(String, nullable=False)
+    thread_id = Column(ForeignKey("threads.id"), nullable=False, index=True)
+    location = Column(Geometry(geometry_type="POINT", srid=4326), nullable=False)
```

Comment:
geom

Reply:
I think it is better to use simple words everybody understands.

Reply:
It's customary with PostGIS to call geometry columns "geom", and we're using it in the other tables too. (I now noticed `nodes` is not, I would change it too).

If you `SELECT` the location as such, you just get some incomprehensible gibberish, the `geom` makes it clear that this is a special column. (Also, it's not good to have more than one geometry column per table, as a lot of the PostGIS functionality depends on one geometry column.)

Reply:
fixed